tests/PersonDialogTest.java

```diff
@@ -0,0 +1,204 @@
+
+        // Test person is added
+        window.table().requireRowCount(1);
+
+    }
+
+    /**
+     * This tests that a person cannot be added with improper characters.
+     */
+
+    @Test
+    void addPersonInvalid() {
+        // Test when user exits with OK
+        window.button("add").click();
+        DialogFixture dialog = window.dialog();
+
+        dialog.textBox("firstName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_1).releaseKey(VK_SHIFT)
+            .pressAndReleaseKeys(VK_2, VK_2, VK_4, VK_5, VK_6);
+        
+        dialog.textBox("lastName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_M).releaseKey(VK_SHIFT)
+            .pressAndReleaseKeys(VK_9, VK_7);
+        
+        // Click 'OK'
+        dialog.button(JButtonMatcher.withText("OK")).click();
+
+        // Test person is not added
+        window.table().requireRowCount(0);
+
+    }
+
+
+    @Test
+    void addPersonNullFirst() {
+        // Test when user exits with OK
+        window.button("add").click();
+        DialogFixture dialog = window.dialog();
+
+        dialog.textBox("firstName").pressKey(VK_SPACE);
+
+        // Click 'OK'
+        dialog.button(JButtonMatcher.withText("OK")).click();
+
+        // Test person is not added
+        window.table().requireRowCount(0);
+
+    }
+
+
+    @Test
+    void addPersonMixedInvlaid() {
+        // Test first name valid and lastname invalid when user exits with OK
+        window.button("add").click();
+        DialogFixture dialog = window.dialog();
+
+        dialog.textBox("firstName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_M).releaseKey(VK_SHIFT)
+            .pressAndReleaseKeys(VK_U, VK_F, VK_F, VK_I, VK_N);
+        dialog.textBox("lastName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_M).releaseKey(VK_SHIFT)
+            .pressAndReleaseKeys(VK_9, VK_7);
+
+        // Click 'OK'
+        dialog.button(JButtonMatcher.withText("OK")).click();
+
+        // Test person is not added
+        window.table().requireRowCount(0);
+
+    }
+
+    // @Test
+    // void addPersonMixedInvlaid2() {
+    //     // Test when user exits with OK
+    //     window.button("add").click();
+    //     DialogFixture dialog = window.dialog();
+
+    //     dialog.textBox("firstName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_1).releaseKey(VK_SHIFT)
+    //         .pressAndReleaseKeys(VK_2, VK_2, VK_4, VK_5, VK_6);
+    //     // dialog.textBox("lastName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_M).releaseKey(VK_SHIFT)
+    //     //     .pressAndReleaseKeys(VK_A, VK_N);
+
+    //     // Click 'OK'
+    //     dialog.button(JButtonMatcher.withText("OK")).click();
+
+    // }
+
+    @Test
+    void addPersonNull() {
+        // Test when user exits with OK
+        window.button("add").click();
+        DialogFixture dialog = window.dialog();
+
+        // Click 'OK'
+        dialog.button(JButtonMatcher.withText("OK")).click();
+
+        // Test person is not added
+        window.table().requireRowCount(0);
```

Comment:
Same as before, removed addressbookGUI. I am not sure this is null, rather empty strings. I am not sure null would be possible via the GUI. Just rename these to "empty".